### PR TITLE
Chore: clarify need update test cases

### PR DESCRIPTION
**Pull request by mowangdk**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Clearify service need update test cases. the next round cleanup of #117595

#### Which issue(s) this PR fixes:

None

#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note

```
None
#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by mowangdk** [comment]
@thockin PTAL~ the next round of cleanup

**Comment by mowangdk** [comment]
> @thockin PTAL~ the next round of cleanup

BTW, I would like to become a member of the kubernetes community, can you be my first sponsor?

File: staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go [hunk 1]
```diff
@@ -68,6 +68,7 @@ func newService(name string, serviceType v1.ServiceType, tweaks ...serviceTweak)
 		},
 		Spec: v1.ServiceSpec{
 			Type: serviceType,
+			Ports: makeServicePort(v1.ProtocolTCP, 0),
```
Comment (thockin) [inline staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go:71]: 0 is not a valid port number - better to set it to 80 or something

File: staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go [hunk 2]
```diff
@@ -1376,54 +1384,48 @@ func TestNeedsUpdate(t *testing.T) {
...
 			return
 		},
 		expectedNeedsUpdate: false,
 	}, {
 		testName: "If service IPFamilies from single stack to dual stack",
 		updateFn: func() (oldSvc *v1.Service, newSvc *v1.Service) {
-			oldSvc = newService("tcp-service", v1.ServiceTypeLoadBalancer, tweakAddPorts(v1.ProtocolTCP, 22), tweakAddAppProtocol("http"), tweakAddIPFamilies(v1.IPv4Protocol))
-			newSvc = oldSvc.DeepCopy()
-			newSvc.Spec.IPFamilies = []v1.IPFamily{v1.IPv4Protocol, v1.IPv6Protocol}
+			oldSvc = newService("tcp-service", v1.ServiceTypeLoadBalancer, tweakAddIPFamilies(v1.IPv4Protocol))
```
Comment (thockin) [inline staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go:1395]: The tweak should be named "set" not "add"

**Review by thockin — COMMENTED** [review]
nice

File: staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go (hunk 1 above)
Comment (mowangdk) [reply staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go:71]: 0 is the symbol indicating whether the service target port is set or not.

```
func makeServicePort(protocol v1.Protocol, targetPort int) []v1.ServicePort {
	sp := v1.ServicePort{Port: 80, Protocol: protocol}
	if targetPort > 0 {
		sp.TargetPort = intstr.FromInt(targetPort)
	}
	return []v1.ServicePort{sp}
}
```

File: staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go (hunk 2 above)
Comment (mowangdk) [reply staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go:1395]: done

**Comment by danwinship** [comment]
/retitle Chore: clarify need update test cases

File: staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go (hunk 1 above)
Comment (thockin) [reply staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go:71]: Ahh, this was unclear.  I assumed it was the port number.

**Review by thockin — COMMENTED** [review]
Thanks!

/lgtm
/approve

File: staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go [hunk 3]
```diff
@@ -68,6 +68,7 @@ func newService(name string, serviceType v1.ServiceType, tweaks ...serviceTweak)
 		},
 		Spec: v1.ServiceSpec{
 			Type: serviceType,
```
Comment (pacoxu) [inline staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go:70]:
```suggestion
			Type:  serviceType,
```

You can run `./hack/update-gofmt.sh` as well.

Comment (mowangdk) [reply staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go:70]: ok, thx~

**Comment by mowangdk** [comment]
> Thanks!
> 
> /lgtm /approve

Is there anything else I can help you with now that this phase of the clean-up is over?

**Pull request merged**